src/Sylius/Bundle/ResourceBundle/Controller/ResourceController.php

```diff
@@ -420,8 +443,20 @@ public function applyStateMachineTransitionAction(Request $request)
             throw new BadRequestHttpException();
         }
 
-        $this->stateMachine->apply($configuration, $resource);
-        $this->manager->flush();
+        try {
+            $this->resourceUpdater->applyTransitionAndFlush($resource, $configuration, $this->manager);
```

Comment:
Anyway, this method not necessarily uses transition, am I right? So maybe we go with simple ``update`` and that's it...